src/test/kotlin/no/nav/familie/ef/sak/service/steg/BeregnYtelseStegTest.kt

```diff
@@ -0,0 +1,174 @@
+package no.nav.familie.ef.sak.service.steg
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import no.nav.familie.ef.sak.api.beregning.Beløpsperiode
+import no.nav.familie.ef.sak.api.beregning.BeregningService
+import no.nav.familie.ef.sak.api.beregning.Innvilget
+import no.nav.familie.ef.sak.api.beregning.VedtakDto
+import no.nav.familie.ef.sak.api.beregning.VedtakService
+import no.nav.familie.ef.sak.no.nav.familie.ef.sak.repository.behandling
+import no.nav.familie.ef.sak.no.nav.familie.ef.sak.repository.fagsak
+import no.nav.familie.ef.sak.no.nav.familie.ef.sak.økonomi.lagAndelTilkjentYtelse
+import no.nav.familie.ef.sak.no.nav.familie.ef.sak.økonomi.lagTilkjentYtelse
+import no.nav.familie.ef.sak.repository.domain.BehandlingType
+import no.nav.familie.ef.sak.repository.domain.TilkjentYtelse
+import no.nav.familie.ef.sak.service.BehandlingService
+import no.nav.familie.ef.sak.service.TilkjentYtelseService
+import no.nav.familie.ef.sak.util.Periode
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.util.UUID
+
+internal class BeregnYtelseStegTest {
+
+    private val tilkjentYtelseService = mockk<TilkjentYtelseService>(relaxed = true)
+    private val behandlingService = mockk<BehandlingService>()
+    private val beregningService = mockk<BeregningService>()
+    private val vedtakService = mockk<VedtakService>(relaxed = true)
+
+    private val steg = BeregnYtelseSteg(tilkjentYtelseService, behandlingService, beregningService, vedtakService)
+
+    @BeforeEach
+    internal fun setUp() {
+        every { behandlingService.hentAktivIdent(any()) } returns "123"
+        every { behandlingService.finnSisteIverksatteBehandling(any()) } returns UUID.randomUUID()
+    }
+
+    @Test
+    internal fun `revurdering - nye andeler legges til etter forrige andeler`() {
+        val forrigeAndelFom = LocalDate.of(2021, 1, 1)
+        val forrigeAndelTom = LocalDate.of(2021, 3, 31)
+        val nyAndelFom = LocalDate.of(2022, 1, 1)
+        val nyAndelTom = LocalDate.of(2022, 1, 31)
+
+        val slot = slot<TilkjentYtelse>()
+        every { tilkjentYtelseService.opprettTilkjentYtelse(capture(slot)) } answers { firstArg() }
+        every { tilkjentYtelseService.hentForBehandling(any()) } returns
+                lagTilkjentYtelse(listOf(lagAndelTilkjentYtelse(100, forrigeAndelFom, forrigeAndelTom)))
+        every { beregningService.beregnYtelse(any(), any()) } returns listOf(lagBeløpsperiode(nyAndelFom, nyAndelTom))
+
+        utførSteg(BehandlingType.REVURDERING)
+
+        val andeler = slot.captured.andelerTilkjentYtelse
+        assertThat(andeler).hasSize(2)
+        assertThat(andeler[0].stønadFom).isEqualTo(forrigeAndelFom)
```

Comment:
Burde vi "asserte" at kildeBehandlingId er ulike for disse, eller?

Reply:
https://github.com/navikt/familie-ef-sak/pull/820/commits/50b3025f8ebac67985b12b62911fbef5c0838970